tests/com/jvm/external/jar/AddJarManifestEntryTest.java

```diff
@@ -359,26 +391,50 @@ private Path createJarWithAutomaticModuleName(String name) throws IOException {
     return jar;
   }
 
-  private void createJar(Path outputTo, Manifest manifest, Map<String, String> pathToContents) throws IOException {
+  private void createJar(Path outputTo, Manifest manifest, Map<String, String> pathToContents)
+          throws IOException {
+    createJar(outputTo, manifest, pathToContents, false);
+  }
+
+  private void createJar(
+          Path outputTo,
+          Manifest manifest,
+          Map<String, String> pathToContents,
+          boolean setExtendedTimestamps)
+          throws IOException {
     try (OutputStream os = Files.newOutputStream(outputTo);
          ZipOutputStream zos = new JarOutputStream(os)) {
 
       if (manifest != null) {
         manifest.getMainAttributes().put(Attributes.Name.MANIFEST_VERSION, "1.0");
         ZipEntry manifestEntry = new ZipEntry(JarFile.MANIFEST_NAME);
+
         zos.putNextEntry(manifestEntry);
         manifest.write(zos);
       }
 
       for (Map.Entry<String, String> entry : pathToContents.entrySet()) {
         ZipEntry ze = new ZipEntry(entry.getKey());
+        if (setExtendedTimestamps) {
+          ze.setLastModifiedTime(FileTime.from(Instant.now()));
+        }
         zos.putNextEntry(ze);
         zos.write(entry.getValue().getBytes(UTF_8));
         zos.closeEntry();
       }
     }
   }
 
+  private void writeJarInTimezone(Path inputJar, Path outputJar, String timeZone) throws IOException {
+    final TimeZone previousTimeZone = TimeZone.getDefault();
+    TimeZone.setDefault(TimeZone.getTimeZone(timeZone));
```

Comment:
Changing the global time zone doesn't feel nice. Happy to discuss alternative ways of testing for the issue.

Reply:
We have complete control over `AddJarManifestEntry`. We can make it a Proper Class rather than just a string of static methods, and then we can set the timezone via a setter in the tests, and just default it to something sensible in the default case.

If you're wary of doing that, I can prep a PR for it.

Reply:
Having said that, the output of your build should really be the same no matter which timezone you run it in.

Reply:
The issue is that we don't have a way to inject the `TimeZone` or a provider for it into `ZipEntry`. It will use `TimeZone.getDefault()` under the hood.

Using a JUnit rule for setting and re-setting the time zone would be a bit better than what I have now, otherwise I'm out of ideas.

Reply:
I can prep a PR to provide some accessors and mutators. It's not a big change.

Reply:
PR ready: https://github.com/bazelbuild/rules_jvm_external/pull/744

The idea here is we can now add a setter for the `TimeZone` on `AddJarManifestEntry` for use in our tests. Feel free to either pull those changes into this PR, or wait until we merge it and rebase.